crates/noirc_evaluator/src/ssa/anchor.rs

```diff
@@ -168,37 +183,43 @@ impl Anchor {
         ctx: &SsaContext,
         item: &MemItem,
         op: &Operation,
-    ) -> Option<CseAction> {
+    ) -> Result<Option<CseAction>, RuntimeErrorKind> {
         let (array_id, index, is_load) = Anchor::get_mem_op(op);
         if let Some(b_value) = ctx.get_as_constant(index) {
             match item {
                 MemItem::Const(p) | MemItem::ConstLoad(p) => {
                     let a = self.get_mem_map(array_id);
                     let b_idx = b_value.to_u128() as usize;
+                    if b_idx >= a.len() {
```

Comment:
Same suggestion here.

Since this length handling code is somewhat repeated perhaps we can have a helper for it `fn get_elem(&self, index: ...) -> Result<..., RuntimeError>`

Reply:
Done.
I created function helper for this, which does not help much for the code size due to mutability, but it makes the code more readable.